src/rules/no-object-math/rule.spec.ts

```diff
@@ -8,26 +8,273 @@ const messageId = "object-math-violation";
+			expect(output).toMatchInlineSnapshot(`
+				"const v1 = new Vector2(1, 2);
+				const v2 = new Vector2(3, 4);
+				const v3 = new Vector2(5, 6);
+				const result = v3.add(v1.add(v2));"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const pos = new Vector3(1, 2, 3);
+			const result = pos * 2 + new Vector3(1, 1, 1);
+		`,
+		errors: [{ messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const pos = new Vector3(1, 2, 3);
+				const result = new Vector3(1, 1, 1).add(pos.mul(2));"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const v1 = new Vector2(10, 20);
+			const v2 = new Vector2(5, 5);
+			const complex = v1 - v2 * 2;
+		`,
+		errors: [{ messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const v1 = new Vector2(10, 20);
+				const v2 = new Vector2(5, 5);
+				const complex = v1.sub(v2.mul(2));"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const cf = new CFrame();
+			const vector = new Vector3(1, 2, 3);
+			const combined = cf * vector + vector;
+		`,
+		errors: [{ messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const cf = new CFrame();
+				const vector = new Vector3(1, 2, 3);
+				const combined = vector.add(cf.mul(vector));"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const v1 = new Vector2(1, 2);
+			const v2 = new Vector2(3, 4);
+			const v3 = new Vector2(5, 6);
+			const result = v1 + v2 * v3;
+		`,
+		errors: [{ messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const v1 = new Vector2(1, 2);
+				const v2 = new Vector2(3, 4);
+				const v3 = new Vector2(5, 6);
+				const result = v1.add(v2.mul(v3));"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const v1 = new Vector2(10, 10);
+			const v2 = new Vector2(2, 2);
+			const v3 = new Vector2(3, 3);
+			const result = v1 / v2 + v3;
+		`,
+		errors: [{ messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const v1 = new Vector2(10, 10);
+				const v2 = new Vector2(2, 2);
+				const v3 = new Vector2(3, 3);
+				const result = v3.add(v1.div(v2));"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const v = new Vector2(10, 10);
+			const result = v * 2 + 3 * 4;
+		`,
+		errors: [{ messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const v = new Vector2(10, 10);
+				const result = v.mul(2).add(3 * 4);"
+			`);
+		},
+	},
+	{
+		code: unindent`
+			const a = new Vector2(1, 1);
+			const b = new Vector2(2, 2);
+			const c = new Vector2(3, 3);
+			const d = new Vector2(4, 4);
+			const result = a + b * c - d;
+		`,
+		errors: [{ messageId }, { messageId }, { messageId }],
+		output: output => {
+			expect(output).toMatchInlineSnapshot(`
+				"const a = new Vector2(1, 1);
+				const b = new Vector2(2, 2);
+				const c = new Vector2(3, 3);
+				const d = new Vector2(4, 4);
+				const result = d.sub(a.add(b.mul(c)));"
```

Comment:
This doesn't seem right - something is being subtracted from d, instead of d being subtracted from something else.